lib/Rex/Commands/File.pm

```diff
@@ -357,6 +372,19 @@ sub file {
     }
 
     return \@ret;
+  } 
+  elsif ( my @exp = $file =~ m/^((?:\\.|[^\\\{])*)({(?:\\.|[^\\\*?\[])*})((?:\\.|[^\\\}])*)$/ ) {
+    if ( $exp[1] =~ /{[^},]*,[^}]*}/ ) {
+      my @ret;
+      # quote braces to keep macros
+      $exp[1] =~ s/({[^,}]*)}/\\$1\\}/g;
+  
+      for my $f ( glob $exp[1] ) {
+          push ( @ret, file( "$exp[0]$f$exp[2]", @options) );
+      }
+
+      return \@ret;
+    }
```

Comment:
El Dimarts 18 Agost 2015, a les 13:19:03, Ferenc Erki va escriure:
> > @@ -357,6 +372,19 @@ sub file {
> >      }
> >  
> >      return \@ret;
> > +  } 
> > +  elsif ( my @exp = $file =~ m/^((?:\\.|[^\\\{])*)({(?:\\.|[^\\\*?\[])*})((?:\\.|[^\\\}])*)$/ ) {
> > +    if ( $exp[1] =~ /{[^},]*,[^}]*}/ ) {
> > +      my @ret;
> > +      # quote braces to keep macros
> > +      $exp[1] =~ s/({[^,}]*)}/\\$1\\}/g;
> > +  
> > +      for my $f ( glob $exp[1] ) {
> > +          push ( @ret, file( "$exp[0]$f$exp[2]", @options) );
> > +      }
> > +
> > +      return \@ret;
> > +    }
> 
> I have a hard time understanding what this code branch does, especially the regex in the condition. Can you explain it, please? I probably miss a corner case you found, because I expected the solution would be simpler, something like:
> ```perl
> my @files_after_glob = glob($file);
> 
> if ( scalar @files_after_glob > 1 ) {
>     Rex::Logger::debug( "Files after glob: " . join ', ', @files_after_glob);
>     my $ret = file( \@files_after_glob, @options );
>     return $ret;
> }
> ```

Sorry for the late answer. I was on holidays and my time on the computer was for the upcoming YAPC::Europe on Granada and its view on Lanyrd <G>

My assumption is that the machine that runs the glob is not the one with the path to be tested. The first regex checks that the expression that we pass to glob don't have wildcards the second is to be sure that glob will return more than one path